libtransmission/peer-msgs.cc

```diff
@@ -2113,17 +2115,26 @@ size_t tr_peerMsgsImpl::max_available_reqs() const
 tr_peerMsgs::tr_peerMsgs(
     tr_torrent const& tor,
     std::shared_ptr<tr_peer_info> peer_info_in,
-    tr_interned_string user_agent,
+    tr_peer_id_t peer_id,
     bool connection_is_encrypted,
     bool connection_is_incoming,
     bool connection_is_utp)
     : tr_peer{ tor }
     , peer_info{ std::move(peer_info_in) }
-    , user_agent_{ user_agent }
+    , peer_id_{ peer_id }
     , connection_is_encrypted_{ connection_is_encrypted }
     , connection_is_incoming_{ connection_is_incoming }
     , connection_is_utp_{ connection_is_utp }
 {
+    auto client = tr_interned_string{};
+    bool peer_id_is_empty = std::all_of(peer_id.begin(), peer_id.end(), [](int i) { return i == 0; });
+    if (!peer_id_is_empty)
```

Comment:
If you are still up for more review changes, this can be simplified:

```suggestion
    if (peer_id != tr_peer_id_t{})
```